prover/src/gpu/metal/mod.rs

```diff
@@ -1,23 +1,29 @@
-//! This module contains GPU acceleration logic for Apple Silicon devices. For now the
-//! logic is limited to GPU accelerating RPO 256 trace commitments.
+//! This module contains GPU acceleration logic for Apple Silicon devices.
+//! For now, the logic is limited to GPU accelerating trace and constraint commitments,
+//! using the RPO 256 or RPX 256 hash functions.
+mod tests;
```

Comment:
Related to the previous comment: we can apply `#[cfg(test)]` here.

Also, I would move this to line ~33 (i.e., put this after `use` statements.